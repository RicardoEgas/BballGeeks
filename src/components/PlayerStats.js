import React, { useState, useEffect } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import { useParams } from 'react-router';

function PlayerStats({ selectedPlayerId }) {
  const { season } = useParams();
  const [playerStats, setPlayerStats] = useState(null);

  useEffect(() => {
    const fetchPlayerStats = async () => {
      if (selectedPlayerId !== null) {
        try {
          const response = await axios.get('https://www.balldontlie.io/api/v1/season_averages', {
            params: {
              season,
              player_ids: [selectedPlayerId],
            },
          });

          if (response.data.data.length > 0) {
            setPlayerStats(response.data.data[0]);
          } else {
            setPlayerStats(null);
          }
        } catch (error) {
          setPlayerStats(null);
        }
      }
    };

    fetchPlayerStats();
  }, [selectedPlayerId, season]);

  return (
    <div>
      <h2> Player Stats </h2>
      {playerStats !== null ? (
        <pre>{JSON.stringify(playerStats, null, 2)}</pre>
      ) : (
        <p>No player stats available for the selected player.</p>
      )}
    </div>
  );
}

PlayerStats.propTypes = {
  selectedPlayerId: PropTypes.number.isRequired,
};

export default PlayerStats;
