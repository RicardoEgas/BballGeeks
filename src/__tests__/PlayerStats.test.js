import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PlayerStats from '../components/PlayerStats';

describe('PlayerStats component', () => {
  it('should render "No player stats available" message when playerStats is null', () => {
    render(<PlayerStats selectedPlayerId={1} playerStats={null} />);
    const noStatsMessage = screen.getByText(/No player stats available for the selected player/i);
    expect(noStatsMessage).toBeInTheDocument();
    expect(noStatsMessage).toMatchSnapshot();
  });
});
