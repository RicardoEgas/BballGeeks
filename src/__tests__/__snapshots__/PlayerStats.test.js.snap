// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStats component should render "No player stats available" message when playerStats is null 1`] = `
<p>
  No player stats available for the selected player.
</p>
`;
